### PR TITLE
New linter and upgrades

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Adding new linters. 
- tparallel, which ensures that t.Parallel is applied
- whitespace, which reports unnecessary whitespace usage

## Short description of the changes

- upgraded linter
- introduced new linters
- fixed code

### CODE REVIEW

File: component/async/kafka/group/group_test.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func TestNew(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
```
Comment (teivah) [inline component/async/kafka/group/group_test.go]: We're missing one `tst := tt`

File: component/http/component_test.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ func Test_createHTTPServer(t *testing.T) {
 }
 
 func TestBuilder_WithShutdownGracePeriod(t *testing.T) {
+	t.Parallel()
```
Comment (teivah) [inline component/http/component_test.go:75]: Same

File: component/http/handler_test.go [hunk 3]
```diff
@@ -176,6 +180,7 @@ func Test_handleSuccess(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
```
Comment (teivah) [inline component/http/handler_test.go:185]: Same

File: component/http/component_test.go (hunk 2 above)
Comment (mantzas) [reply component/http/component_test.go:75]: done

File: component/http/handler_test.go (hunk 3 above)
Comment (mantzas) [reply component/http/handler_test.go:185]: done

**Comment by mantzas** [comment]
There is some flakiness in our test most probably introduced by the parallel execution and the usage of the global tracer.

**Review by teivah — APPROVED** [review]

**Review by tpaschalis — APPROVED** [review]
Nice one!

**Pull request merged**